app/controllers/staff/activity_forms_controller.rb

```diff
@@ -31,14 +31,42 @@ def update
     @activity = Activity.find(params[:activity_id])
     authorize @activity
 
-    @activity.assign_attributes(activity_params)
+    # Update activity dates, after checking they are valid via format_date
+    update_activity_dates
+    # Update all attributes except dates
+    @activity.assign_attributes(activity_params.select { |param| param.match(/^((?!date).)*$/) })
```

Comment:
I suggest flipping this `select` to a `reject` and using the same regex you're using in `update_activity_dates`.